general/ArgumentParser.cpp

```diff
@@ -34,123 +28,25 @@
-     fprintf(stderr, "     %s  -h/--help       display this text.\n", file.c_str());
-#ifdef _JPS_AS_A_SERVICE
-     fprintf(stderr, "     %s  --as-a-service -p <port nr> runs jps as a service at port <port nr>.\n", file.c_str());
-#endif
-     exit(EXIT_SUCCESS);
+const fs::path &ArgumentParser::IniFilePath() const {
+    return iniFilePath;
 }
 
-ArgumentParser::ArgumentParser(Configuration* config)
-{
-     _config = config;
+std::tuple<ArgumentParser::Execution, int>
+ArgumentParser::Parse(int argc, char *argv[]) {
+    try {
+        app.parse(argc, argv);
+        // Silence warnigns about iniFilePathOpt declared as member but never
+        // used. iniFilePathOpt is a member to keep the declaration of all
+        // options in the header file. If there was no iniFilePathOpt member the
+        // add_option(...) call would need to be done the constructor.
+        (void)iniFilePathOpt;
+        if (auto msg = CLI::ExistingFile(iniFilePath.string()); !msg.empty()) {
+            throw CLI::ValidationError(msg);
 }
-
-ArgumentParser::~ArgumentParser(){}
-
-bool ArgumentParser::ParseArgs(int argc, char** argv)
-{
-     //special case of the default configuration ini.xml
-     if (argc==1) {
-          Log->Write(
-                     "INFO: \tTrying to load the default configuration from the file <ini.xml>");
-     // first logs will go to stdout
-          Log->Write("----\nJuPedSim - JPScore\n");
-          Log->Write("Current date   : %s %s", __DATE__, __TIME__);
-          Log->Write("Version        : %s", JPSCORE_VERSION);
-          // Log->Write("Compiler       : %s (%s)", true_cxx.c_str(), true_cxx_ver.c_str());
-          Log->Write("Commit hash    : %s", GIT_COMMIT_HASH);
-          Log->Write("Commit date    : %s", GIT_COMMIT_DATE);
-          Log->Write("Branch         : %s\n----\n", GIT_BRANCH);
-
-          IniFileParser* p = new IniFileParser(_config);
-          if (!p->Parse("ini.xml")) {
-               Usage(argv[0]);
+    } catch (const CLI::ParseError &e) {
+        return {Execution::ABORT, app.exit(e)};
           }
-          return true;
-     }
-
-     std::string argument = argv[1];
-
-     if (argument=="-h" || argument=="--help") {
-          Usage(argv[0]);
-          return false;
-     }
-
-     // other special case where a single configuration file is submitted
-     //check if inifile options are given
-     if (argc==2) {
-          std::string prefix1 = "--ini=";
-          std::string prefix2 = "--inifile=";
-
-          if (!argument.compare(0, prefix2.size(), prefix2)) {
-               argument.erase(0, prefix2.size());
-          }
-          else if (!argument.compare(0, prefix1.size(), prefix1)) {
-               argument.erase(0, prefix1.size());
-          }
-          IniFileParser* p = new IniFileParser(_config);
-          bool status = p->Parse(argument);
-          delete p;
-          return status;
-     }
-
-#ifdef _JPS_AS_A_SERVICE //TODO try to avoid macros!
```

Comment:
Hmm this is going to be, ehem... difficult. the removed section contains some additional flags that I cannot test since I cannot compile with _JPS_AS_A_SERVICE. I was under the impression this was defunct. If this is needed, I maybe should try to get it compiling again.

Reply:
`_JPS_AS_A_SERVICE` does not have to be tested. It sounds weird, but it's going to be used in an upcoming project. 
When this happens, we will have to deal with it in another PR.